app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,59 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :owned_post, only: [:edit, :update, :destroy]
```

Comment:
great guard clauses - this is exactly the sort of thing that gets forgotten during tech tests, so well remembered.